Library/Homebrew/extend/ARGV.rb

```diff
@@ -65,10 +65,18 @@ def kegs
           Keg.new(linked_keg_ref.resolved_path)
         elsif dirs.length == 1
           Keg.new(dirs.first)
-        elsif (prefix = (name.include?("/") ? Formulary.factory(name) : Formulary.from_rack(rack)).prefix).directory?
-          Keg.new(prefix)
         else
-          raise MultipleVersionsInstalledError.new(rack.basename)
+          f = if name.include?("/") || File.exist?(name)
```

Comment:
Just for completeness' sake: Same comment about stricter check and maybe a `formula_file?` helper method to avoid duplicate and inconsistent checks.